src/components/molecules/EarthEditor/PropertyPane/PropertyField/URLField/index.tsx

```diff
@@ -37,13 +36,29 @@ const URLField: React.FC<Props> = ({
   linked,
   overridden,
   fileType,
-  assetsData,
+  assets,
+  isAssetsLoading,
+  hasMoreAssets,
+  assetSort,
+  assetSearchTerm,
+  onGetMoreAssets,
+  onCreateAssets,
+  onAssetSort,
+  onAssetSearch,
   onChange,
 }) => {
   const intl = useIntl();
+  const gqlCache = useApolloClient().cache;
```

Comment:
Rule Violation: molecules cannot use apollo directly!! `onAssetModalClose` and impl logic to delete cache in organisms

Reply:
Since AssetContainer is a molecule itself, I wasn't able to move it down into it. I guess it is better to make an AssetContainer organism and move the cache deletion from AssetModal to there. If I do that, I won't have to pass onAssetModalClose prop as you suggested here!

Reply:
Sorry, @rot1024 I remember now why I didn't do this in the first place. If I do as you ask here, I would need to create an AssetContainer organism, add the cache deletion there, and then the only way I can use it here in URLField is to pass the organism as a prop all the way down to here. 
The problem is as you can see, I am using local variables that I am unable to reach until the URLField. In particular `value`.
In saying that, would you say moving cache deletion is impossible? If so, your `onAssetModalClose` option seems the only option

Reply:
OrganismsAssetModal can accept selected value. Organisms components are not prohibited to accept normal props.

`src/components/organisms/Common/AssetModal/index.tsx`:

```
const OrganismsAssetModal = ({ value }: { value?: string }) => {
  // gql calls
  return <AssetMolecules ... />
}
```

Reply:
Yes, but I'll need to pass `OrganismAssetModal` as a prop from (I believe) the `PropertyPane` organism, which means I can't pass it `value` from `URLField`, and to get the `value` in the `PropertyPane` organism would be close to impossible without a lot of work. Am I incorrect in this?

Reply:
How about changing props of molecule PropertyPanel?

```
type Props = {
 assetModal?: (value: string | undefined) => ReactNode
}
```

Reply:
I was REALLY hoping something like that was possible, but I thought it was impossible. Thank you!